src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserController.kt

```diff
@@ -328,6 +331,53 @@ class UserController(
     val groups = userGroupService.getMyAssignableGroups()
     return groups.map { UserGroupDto(it) }
   }
+
+  @PostMapping("/email")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Email address for users",
+    description =
+    """Verified email address for users.  Post version that accepts multiple email addresses.
+        Requires ROLE_MAINTAIN_ACCESS_ROLES or ROLE_MAINTAIN_ACCESS_ROLES_ADMIN.
+    """
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Unable to enable user, the user is not within one of your groups.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_ACCESS_ROLES', 'ROLE_MAINTAIN_ACCESS_ROLES_ADMIN')")
+  suspend fun getUsersEmails(
+    @Parameter(description = "List of usernames.", required = true) @RequestBody
+    usernames: List<String>
+  ) =
+    userSearchService
+      .findUsersByUsernames(usernames)
+      .filter { it.verified }
```

Comment:
Can't we do this filter in the database? Or if not would be better in the service layer as feels like business logic that the controller should not be doing?